### PR TITLE
feat(metrics) - new table for performance metrics indexer

### PR DESCRIPTION
**Pull request by onewland**
Created a new Django model and table for the indexer for performance metrics, which should be functionally and schematically equivalent to release-health metrics.

For SaaS we'll need to set up the configuration in the `getsentry` repo to route this table to a new database created for this feature

### CODE REVIEW

File: src/sentry/sentry_metrics/indexer/models.py [hunk 1]
```diff
@@ -55,3 +60,18 @@ class Meta:
...
         ]
+
+
+class PerfStringIndexer(BaseIndexer):
+    __include_in_export__ = False
+
+    objects = BaseManager(cache_fields=("pk",), cache_ttl=settings.SENTRY_METRICS_INDEXER_CACHE_TTL)  # type: ignore
+
+    class Meta:
+        db_table = "sentry_perfindexer"
```
Comment (wedamija) [inline src/sentry/sentry_metrics/indexer/models.py:71]: This should be `sentry_perfstringindexer`

File: src/sentry/sentry_metrics/indexer/models.py [hunk 2]
```diff
@@ -55,3 +60,18 @@ class Meta:
         constraints = [
             models.UniqueConstraint(fields=["string", "organization_id"], name="unique_org_string"),
         ]
+
+
+class PerfStringIndexer(BaseIndexer):
+    __include_in_export__ = False
+
+    objects = BaseManager(cache_fields=("pk",), cache_ttl=settings.SENTRY_METRICS_INDEXER_CACHE_TTL)  # type: ignore
```
Comment (wedamija) [inline src/sentry/sentry_metrics/indexer/models.py:68]: Since its identical, the manager can go on the base class

**Review by wedamija — APPROVED** [review]
This looks ok in general. Is the plan to have a getsentry pr ready to update the routings, so that the table is created in the right place? Or will ops just create it manually?

**Comment by onewland** [comment]
> This looks ok in general. Is the plan to have a getsentry pr ready to update the routings, so that the table is created in the right place? Or will ops just create it manually?

I think I'll probably put up a getsentry PR for the routings before I merge this one

File: src/sentry/sentry_metrics/indexer/models.py [hunk 3]
```diff
@@ -55,3 +60,18 @@ class Meta:
...
+
+    class Meta:
+        db_table = "sentry_perfindexer"
+        app_label = "sentry"
+        constraints = [
+            models.UniqueConstraint(
+                fields=["string", "organization_id"], name="perf_unique_org_string"
```
Comment (markstory) [inline src/sentry/sentry_metrics/indexer/models.py:75]: Do you ever query by only organization_id?

**Comment by markstory** [comment]
> I think I'll put up a getsentry PR for the routings before I merge this one

Yes, you'll need routing setup before merging this. Also worth pointing out that deploys will not run migrations for the new database automatically. You'll need to add the `RUN_MIGRATIONS` option to the database configuration. See the `usage` connection for an example.

**Comment by onewland** [comment]
getsentry PR: https://github.com/getsentry/getsentry/pull/7602

**Pull request merged**